tests/psp22_contract/Cargo.toml

```diff
@@ -5,7 +5,7 @@ authors = ["Brushfam <green@727.ventures>"]
 edition = "2021"
 
 [dependencies]
-ink = { version = "4.2.1", default-features = false }
+ink = { version = "=4.3.0", default-features = false }
```

Comment:
Request: do not use `=4.3.0` in `Cargo.toml`, it's painful to untangle afterwards when you need to update another dep. Instead, `cargo update -p 4.3.0 ink` or whatever the syntax is.

Reply:
I think it's been de facto standard for us to fix the version of ink!. The reason being the underlying deps were updated randomly and easily broke the build when it was left up to the cargo to figure the "safe" updates.

Reply:
That's what `.lock` is for. This is in `tests/psp22_contract`, so `.lock` will be in effect.

Reply:
That wasn't my experience 🤔 cargo thinks it's safe to bump indirect deps of `ink` because they match in semver, and will for example bump `ink_env` to a version that is not bin compatible with for example 4.2.1.

Also, the psp22 repository uses ink 4.3.0 so it seemed correct to bump it here https://github.com/Cardinal-Cryptography/PSP22/blob/main/Cargo.toml#L14